server/src/main/java/org/opensearch/index/translog/transfer/TranslogTransferManager.java

```diff
@@ -133,19 +133,26 @@ public boolean transferSnapshot(TransferSnapshot transferSnapshot, TranslogTrans
         }
     }
 
-    public boolean downloadTranslog(String primaryTerm, String generation, Path location) throws IOException {
+    public boolean downloadTranslog(String primaryTerm, String generation, Path location, boolean override) throws IOException {
```

Comment:
What's the use case for override here? It looks like it used only in tests. If we really need it, I'd suggest keeping it out of the public API have the existing public API delegate to a package-private method that includes the boolean option.

Reply:
I am also thinking to make override false at the time of restoring data. But until we make the change to the restore flow, I will go with you suggestion above.